### DISCUSSION TITLE
Sanitize HTML using isomorphic-dompurify

### DISCUSSION THREAD

**Discussion by dgallardox** [wpengine/faustjs#1431] [Ideas]
Perhaps unnecessary but always good for client to ensure what is rendered is safe!

```
import DOMPurify from 'isomorphic-dompurify';

export default function sanitizeHTML(unsanitizedHTML) {
  let cleanHTML = DOMPurify.sanitize(unsanitizedHTML, {
    USE_PROFILES: { html: true },
  });
  return cleanHTML;
}

 <div
  dangerouslySetInnerHTML={{
     __html: sanitizeHTML(post.excerpt),
  }}
/>
```

Currently, we are just using the dangerouslySetInnerHTML without any sanitizing to ensure it is safe. Of course, we can generally assume what is coming from the WordPress side is clean but better to be safe :)

**Comment by theodesp** [comment]
Hey @dgallardox thanks. 

Where do you think the framework should intervene here and sanitize the post data since the user has control over the query structure?

I mean you have the option to use whatever library you want whether it is `isomorphic-dompurify` or [Sanitizer](https://developer.mozilla.org/en-US/docs/Web/API/Sanitizer/Sanitizer) as the framework does not have access to the underlying data you want to fetch.

**Comment by theodesp** [comment]
@dgallardox any thoughts here?

**Comment by theodesp** [comment]
I will move this to a discussion to get more community feedback.
